packages/data-context/src/data/ProjectConfigIpc.ts

```diff
@@ -238,6 +240,37 @@ export class ProjectConfigIpc extends EventEmitter {
 
     debug('fork child process %o', { CHILD_PROCESS_FILE_PATH, configProcessArgs, childOptions: _.omit(childOptions, 'env') })
 
+    let isProjectUsingESModules = false
```

Comment:
With this solution should any code be removed? I'm not super familiar with this code, but it seems like we might have some dead references esbuild code if this block takes over that responsibility. You mention a couple of places in your [explanation](https://github.com/cypress-io/cypress/issues/22074#issuecomment-1147097883), can we clean any of those up?

Reply:
I do think we can clean up some of the `esbuild` code (it's not working properly; we could delete it all, if we decide not to go down the esbuild route).

I figured I'd leave it untouched and make the changes as minimal as possible here. Having read your comment, I'm thinking we could probably delete it, and bring it back when re-visiting https://github.com/cypress-io/cypress/issues/22074.

Reply:
I tried removing the `esbuild` code path here: https://github.com/cypress-io/cypress/pull/22118/commits/f1ed50824aa7ebb5b38b439b7403bf70835614d2. We can revisit how best to use esbuild, if users want to use it, in the near future.